### ISSUE TITLE
[Feature Request]: Generate Labels at a starting location on the page

### ISSUE THREAD

**Issue by wgarvin1** [grokability/snipe-it#13960]
### Is your feature request related to a problem? Please describe.

When trying to print labels using the Avery 5520 style, or 30 labels per page, there is no way to select the starting location on the page. This is an issue when you only need to print a few labels and you don't have a full page of labels available. For Example: if you've already printed 2 labels, and need to print more on the page, you can't select the starting location to be the 3rd label/column on the first row of labels for Avery 5520 page. It always starts at the upper left corner of the page. 

### Describe the solution you'd like

I would like a prompt for the first page of labels printed that would allow me to specify the starting location for printing asset labels based on the new label engine types. You would start from the upper left label and go to the bottom right label. It could be a integer number from 0 to max labels per page minus 1 to get the final position of the page. 

### Describe alternatives you've considered

The alternative was to use your APIs to find and build this functionality myself, but it's annoying because the QR Code url has to get generated before it can be accessed. However, using the Generate Labels option under All Assets page doesn't create the QR Code URL correctly and the API will fail to function as the QR Code wasn't generated. I then have to go into the View Asset page and use the Generate Label Button and once completed it will create the label correctly for me to use when printing from my solution.

**Comment by KeeganPBR** [comment]
i am in the same boat, hopefully someone knows a fix.

**Comment by BlackFenix2** [comment]
It would involve modifying the controller page here: https://github.com/snipe/snipe-it/blob/master/tests/Feature/Assets/Ui/BulkEditAssetsTest.php

Its a POST Request so the response is the outputted PDF, i could try a PR to figure offsetting positions.

**Comment by snipe** [comment]
@BlackFenix2 - those are tests - they would not affect the actual UI.

**Comment by jl-stellar** [comment]
I recently ran into the same problem. Looking through the code, it seems that as of v7.1.5, most of the code needed for this feature seems to already be implemented :) basically there is just no UI for this yet.

Until then, the "print starting index" can be set by modifying the file `app/View/Label.php` around line 181:
```
180         if ($template instanceof Sheet) {
181             $template->setLabelIndexOffset($offset ?? 0);
182         }
```
--> Change the `0` to your desired value.

Of course this is not the intended way and I don't know yet if it breaks other things, but in my case it is working.

**Comment by BlackFenix2** [comment]
@snipe I tried running the docker container for the dev server and it keeps returning 404. I tried to submit a PR but the docs just explain how to setup the docker container and nothing after.

**Comment by snipe** [comment]
@BlackFenix2 I’m not really sure how a 404 on your docker would be related to label generation.

**Comment by snipe** [comment]
@jl-stellar I don’t know where we’d put the GUI side of that, since that number will change each time.

**Comment by jl-stellar** [comment]
Yes, I've been wondering that myself when I looked through the code. 3 places came to my mind:
1. In the label settings
2. As a popup/modal after starting the label generation
3. Beside the Actions dropdown where you select "generate labels" 
My favourite would be #3, but I don't know if that is easily done.
#1 is of course not as comfortable for the user since you would have to set this prior to every label generation (and you are bound to forget that sometimes), but should be pretty easy to implement and is still better than editing the source code, so maybe this could be a first step?

**Comment by snipe** [comment]
I believe this was fixed in this PR https://github.com/grokability/snipe-it/pull/16790

**Comment by BlackFenix2** [comment]
@Godmartinz can we reopen or create a new issue? The PR referenced did a row offset within the single label. This feature request was to offset the label position, like if you already used up 4-5 of an Avery sheet.

I tried to submit a PR, but scaffolding the device server was impossible for me (using docker image)
